### PR TITLE
[Monitoring] Emit prometheus metrics for hostname and git sha

### PR DESCRIPTION
**Pull request by sitalkedia**
### Description

This will help us map peer_id to hostname as well as gives us release split information in real-time.

### Test Plan
Existing UTs

### CODE REVIEW

File: crates/node-resource-metrics/src/collectors/basic_node_info_collector.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+use once_cell::sync::Lazy;
+use prometheus::{
+    core::{Collector, Desc},
+    proto::MetricFamily,
+    register_int_gauge_vec, IntGaugeVec,
+};
+
+/// Current host name
+pub static HOST_NAME: Lazy<IntGaugeVec> = Lazy::new(|| {
+    register_int_gauge_vec!("aptos_node_host_name", "A no-op counter value", &[
```
Comment (ibalajiarun) [inline crates/node-resource-metrics/src/collectors/basic_node_info_collector.rs:13]: Two problems here:
1. We are double registering. This macro already registers the counter, and its being registered below in `register_basic_node_info_collectors`. We should not register here, but use a macro that doesn't register.
2. I don't see where `set(...)` is actually called on `HOST_NAME`. We can actually do this inside `collect()`.

File: crates/node-resource-metrics/src/collectors/basic_node_info_collector.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        "host_name"
+    ])
+    .unwrap()
+});
+
+/// Git hash of the current release.
+pub static NODE_RELEASE_GIT_HASH: Lazy<IntGaugeVec> = Lazy::new(|| {
```
Comment (ibalajiarun) [inline crates/node-resource-metrics/src/collectors/basic_node_info_collector.rs:20]: Same as above.

**Review by ibalajiarun — CHANGES_REQUESTED** [review]

File: crates/node-resource-metrics/src/collectors/basic_node_info_collector.rs (hunk 1 above)
Comment (sitalkedia) [reply crates/node-resource-metrics/src/collectors/basic_node_info_collector.rs]: >We are double registering. This macro already registers the counter, and its being registered below in register_basic_node_info_collectors. We should not register here, but use a macro that doesn't register.

Good point - changed it. 

> I don't see where set(...) is actually called on HOST_NAME. We can actually do this inside collect()

Hah, my bad. I had changes locally which did that but missed to add it to the PR. Updated now.

File: crates/node-resource-metrics/src/collectors/basic_node_info_collector.rs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            .lock()
+            .host_name()
+            .unwrap_or_else(|| String::from(UNKNOW_LABEL));
+
+        let host_name_metrics =
+            ConstMetric::new_gauge(self.host_name.clone(), 1.0, Some(&[host_name])).unwrap();
+
+        let git_hash = aptos_build_info::get_git_hash();
+        let release_metrics =
+            ConstMetric::new_gauge(self.release.clone(), 1.0, Some(&[git_hash])).unwrap();
```
Comment (ibalajiarun) [inline crates/node-resource-metrics/src/collectors/basic_node_info_collector.rs:76]: what do you think about adding another label indicating if git is dirty (i.e. the files were modified)? shadow_rs can give us that info. We dont collect it in BQ but I always had that back of my mind.

File: crates/node-resource-metrics/src/collectors/basic_node_info_collector.rs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    core::{Collector, Desc, Describer},
+    proto::MetricFamily,
+    Opts,
+};
+use std::sync::Arc;
+use sysinfo::{RefreshKind, System, SystemExt};
+
+const BASIC_NODE_INFO_METRICS_COUNT: usize = 2;
+const RELEASE_GIT_HASH_LABEL: &str = "release_git_hash";
+const NODE_HOST_NAME_LABEL: &str = "host_name";
```
Comment (ibalajiarun) [inline crates/node-resource-metrics/src/collectors/basic_node_info_collector.rs:17]: naming nit: I think hostname is a single word?

**Review by ibalajiarun — APPROVED** [review]
LGTM overall.

File: crates/node-resource-metrics/src/collectors/basic_node_info_collector.rs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+impl BasicNodeInfoCollector {
+    fn new() -> Self {
+        let system = Arc::new(Mutex::new(System::new_with_specifics(RefreshKind::new())));
+
+        let release = Opts::new(RELEASE_GIT_HASH_LABEL, "Release git hash.")
+            .namespace(NAMESPACE)
+            .variable_label(GIT_HASH_LABEL)
+            .describe()
+            .unwrap();
+        let host_name = Opts::new(NODE_HOST_NAME_LABEL, "Host name.")
```
Comment (davidiw) [inline crates/node-resource-metrics/src/collectors/basic_node_info_collector.rs:39]: nit hostname not host name: https://en.wikipedia.org/wiki/Hostname

**Review by davidiw — APPROVED** [review]

File: crates/node-resource-metrics/src/collectors/basic_node_info_collector.rs (hunk 5 above)
Comment (JoshLind) [reply crates/node-resource-metrics/src/collectors/basic_node_info_collector.rs:39]: I think part of the issue is that sys_info uses `host_name`: https://docs.rs/sysinfo/latest/sysinfo/struct.System.html 😄

Comment (sitalkedia) [reply crates/node-resource-metrics/src/collectors/basic_node_info_collector.rs:39]: I am hoping this is accurate enough for our use case. Will see :)

File: crates/node-resource-metrics/src/collectors/basic_node_info_collector.rs (hunk 3 above)
Comment (sitalkedia) [reply crates/node-resource-metrics/src/collectors/basic_node_info_collector.rs:76]: Can you point me to the function that can provide us this info? May be I can add that as a followup.

**Pull request merged**